### PR TITLE
Add CI workflow

### PR DESCRIPTION
**Pull request by declan-warn**
Github action to replace Travis CI

### CODE REVIEW

**Comment by declan-warn** [comment]
@danieldelcore it seems to be working now :) 

Interesting it looks like the Travis pipeline wasn't catching linting errors? 🤔 

Also I think you will need to make this new workflow a required one. It isn't preventing merges.

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
```
Comment (danieldelcore) [inline .github/workflows/node.js.yml:4]: Can we rename this workflow to `Test` and update the file name too please 🙏

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
```
Comment (danieldelcore) [inline .github/workflows/node.js.yml:2]: Let's delete these comments, if you dont mind

Comment (draperd) [reply .github/workflows/node.js.yml:2]: What's the rationale for removing the comments?

Comment (danieldelcore) [reply .github/workflows/node.js.yml]: Just thought that these were comments brought across from a template. Happy to keep them if we think we should 😄 
Was a nitpick from me really

Comment (declan-warn) [reply .github/workflows/node.js.yml]: It was left over from the template. I guess some sort of comment could be helpful though 🤔

Comment (declan-warn) [reply .github/workflows/node.js.yml]: I've added a comment that better captures the intent of the workflow

**Comment by danieldelcore** [comment]
Here's a PR for the lint errors breaking our beautiful new Action

https://github.com/atlassian/extract-react-types/pull/191

**Comment by declan-warn** [comment]
@danieldelcore seems like there was one more linting error that `yarn lint:fix` wasn't catching. I made a PR to fix it here and can confirm that's the last of them :) https://github.com/atlassian/extract-react-types/pull/193

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (danieldelcore) [reply .github/workflows/node.js.yml]: Amazing! Yeah sorry if that was vague feedback on my part!

**Comment by danieldelcore** [comment]
Great work on this one mate! Let's merge it!

**Pull request merged**